pkg/vm/emit/emit.go

```diff
@@ -59,14 +59,18 @@ func Int(w *io.BinWriter, i int64) {
 		val := opcode.Opcode(int(opcode.PUSH1) - 1 + int(i))
 		Opcodes(w, val)
 	default:
-		buf := bigint.ToPreallocatedBytes(big.NewInt(i), make([]byte, 0, 32))
-		// l != 0 becase of switch
-		padSize := byte(8 - bits.LeadingZeros8(byte(len(buf)-1)))
-		Opcodes(w, opcode.PUSHINT8+opcode.Opcode(padSize))
-		w.WriteBytes(padRight(1<<padSize, buf))
+		bigInt(w, big.NewInt(i))
 	}
 }
 
+func bigInt(w *io.BinWriter, n *big.Int) {
+	buf := bigint.ToPreallocatedBytes(n, make([]byte, 0, 32))
+	// l != 0 becase of switch
+	padSize := byte(8 - bits.LeadingZeros8(byte(len(buf)-1)))
```

Comment:
We now use it outside of the `switch` as a separate function, so `// l != 0 becase of switch` comment is not relevant. Is it OK to have padSize = 0 here?

Reply:
The problem is that `len(buf)` can be `0` for `n == 0`, so yes, it is a problem.

Reply:
Push?